src/transformers/generation/utils.py

```diff
@@ -387,6 +387,8 @@ def prepare_inputs_for_generation(
             past_length = past_key_values[0][0].shape[2] if past_key_values is not None else 0
             cache_position = torch.arange(past_length, input_ids.shape[1], dtype=torch.long, device=input_ids.device)
 
+        assert cache_position is not None
```

Comment:
```suggestion
```

(We don't want `assert` in `transformers`)

Reply:
Reverted, but note that cache_position can be None